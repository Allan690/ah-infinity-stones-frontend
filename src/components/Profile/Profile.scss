$font-stack: "Segoe UI", Arial, sans-serif;
$primary-color: #86c232;
$secondary-color: #da2687;
$another-color: #f0ff0f;

// %message-shared is extended below.
// $npm run build-css will print in generated css
%message-shared {
  border: 4px solid #ccc;
  padding: 10px;
  color: #333;
}

// %equal-heights is not extended.
%equal-heights {
  display: flex;
  flex-wrap: wrap;
}

body {
  font: $font-stack; 

}

.app-header {
  background-color: $primary-color;
}

.app-link {
  color: $primary-color;
  text-decoration: none;
}

.spinner {
  color: $another-color;
}

.message {
  @extend %message-shared;
}

.success {
  @extend %message-shared;
  border-color: #86c232;
}

.error {
  @extend %message-shared;
  border-color: red;
}

.warning {
  @extend %message-shared;
  border-color: yellow;
}

// normalize adDed
section {
  margin: 10px;
  text-align: left;
  color: #282c34;
}
.dark {
  color: #282c34;
}

.theme{
  color: $primary-color !important;
}

/*------------------------------------------------------------------------------*/
/* BUTTONS */
input[type="submit"],
input[type="reset"],
input[type="button"],

.button {
  background-color: #ddd;
  border: 0;
  border-radius: 4px !important;
  color: #777 !important;

  cursor: pointer;
  display: inline-block;
  font-size: 0.9em;

  height: 4em;
  line-height: 4em;
  padding: 0 3em;
  text-align: center;
  text-decoration: none;
  text-transform: none !important;
  white-space: nowrap;
  margin-bottom: 3px;
}

// button:focus {
//   outline: 0;
// }

input[type="submit"]:hover,
input[type="reset"]:hover,
input[type="button"]:hover,
// button:hover,
.button:hover {
  color: #888 !important;
  background-color: #ccc;
}

/*COLORED buttons-------*/
.button-error {
  /**/
  color: #bbb !important;
  background-color: #c00;
}
.button-error:hover {
  color: #ccc !important;
  background-color: #f00;
}

.button-theme {
  color: #bbb !important;
  background-color: #628835;
  margin-bottom: 26px;
}
.button-theme:hover {
  color: #fff !important;
  background-color: #88c03e;
}

.theme {
  color: #628835;
}

// .subscribe{
//   color: #fff !important;
//   background-color: #474b4f !important;
// }

/*------------------------------------------------------------------------------*/
/* FORMS */
input[type="text"],
input[type="password"],
select,
textarea {
  /**/
  width: 100%;
  padding: 13px;
  border: 1px solid #ccc;
  border-radius: 4px; font-size: 14px !important;

  margin-top: 0px;
  margin-bottom: 26px;
}

textarea {
  resize: vertical; font-family: 'Courier New', Courier, monospace;
}

/*------------------------------------------------------------------------------*/
/* IMG */
img.logo {
  width: 50px;
  height: 50px;

  border-radius: 50%;
  padding: 1px;
  background-color: #fff;
  border: 1px solid #ddd;
}

img.icon-small {
  width: 50px;
  height: 50px;

  border-radius: 50%;
  padding: 1px;
  background-color: #fff;
  border: 1px solid #ddd;
}

img.icon-user {
  width: 50px;
  height: 50px;

  border-radius: 50%;
  padding: 1px;
  background-color: #fff;
  border: 1px solid #ddd;
  margin-top: 13px;
}

img.icon-big {
  width: 100px;
  height: 100px;

  border-radius: 50%;
  padding: 1px;
  background-color: #fff;
  border: 1px solid #ddd;
  margin-top: 13px;
}
img.icon-small-sq {
  width: 50px;
  height: 50px;

  border-radius: 3px;
  padding: 1px;
  background-color: #fff;
  border: 1px solid #ddd;
}

img.icon-big-sq {
  width: 150px;
  height: 150px;

  border-radius: 3px;
  padding: 1px;
  border: 1px solid #ddd;
  margin-top: 13px;
}

.login-footer {
  font-size: 0.8em;
}

.m-profile {
  margin-bottom: 4px;
}
.m-profile > label {
  font-size: 14px;
  color: #628835;
}

.alert-msg {
  font-size: 1.5em !important; font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.profile-list > i {
  color: #628835;
}

a.colored:link,
a.colored:visited {
  color: #777;
  text-decoration: none;
}
a.colored:hover {
  color: #628835;
  text-decoration: underline;
}

a.socialMediaIcon:link,
a.socialMediaIcon:visited {
  color: #777;
  text-decoration: none;
}
a.socialMediaIcon:hover {
  color: #628835;
  text-decoration: underline;
}
a {
  text-decoration: none;
  color: #777;
}
a:hover{
  text-decoration: none;
}
