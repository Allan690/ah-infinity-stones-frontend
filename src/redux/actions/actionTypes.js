export const REQUEST_LOGIN = 'REQUEST_LOGIN';
export const RECEIVE_LOGIN = 'RECEIVE_LOGIN';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const SIGNUP_REQUEST = 'SIGNUP_REQUEST';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const SOCIAL_LOGIN_REQUEST = 'SOCIAL_LOGIN_REQUEST';
export const SOCIAL_LOGIN_SUCCESS = 'SOCIAL_LOGIN_SUCCESS';
export const SOCIAL_LOGIN_ERROR = 'SOCIAL_LOGIN_ERROR';


export const REQUEST_INCREMENT = 'REQUEST_INCREMENT';
export const RECEIVE_INCREMENT = 'RECEIVE_INCREMENT';

export const PASSWORD_RESET_START = 'PASSWORD_RESET_START';
export const PASSWORD_RESET_COMPLETE = 'PASSWORD_RESET_COMPLETE';
export const PASSWORD_RESET_ERROR = 'PASSWORD_RESET_ERROR';


export const PASSWORD_RESET_UPDATE_START = 'PASSWORD_RESET_UPDATE_START';
export const PASSWORD_RESET_UPDATE_COMPLETE = 'PASSWORD_RESET_UPDATE_COMPLETE';
export const PASSWORD_RESET_UPDATE_ERROR = 'ASSWORD_RESET_UPDATE_ERROR';
