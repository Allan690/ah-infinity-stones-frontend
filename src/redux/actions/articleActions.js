import {
  CREATE_ARTICLE, CREATE_ARTICLE_SUCCESS, CREATE_ARTICLE_FAILURE,
  GET_ALL_ARTICLES_REQUEST, GET_ALL_ARTICLES_SUCCESS, GET_ALL_ARTICLES_FAILURE,
  GET_ONE_ARTICLE_REQUEST, GET_ONE_ARTICLE_SUCCESS, GET_ONE_ARTICLE_FAILURE,
  EDIT_ARTICLE, UPDATE_ARTICLE, DELETE_ARTICLE,
} from './actionTypes';


export const createArticleAction = article => ({
  type: CREATE_ARTICLE,
  payload: article,
});
export const createArticleSuccessAction = article => ({
  type: CREATE_ARTICLE_SUCCESS,
  payload: article,
});
export const createArticleFailureAction = error => ({
  type: CREATE_ARTICLE_FAILURE,
  payload: error,
});

export const getAllArticlesRequestAction = () => ({
  type: GET_ALL_ARTICLES_REQUEST,
});

export const getAllArticlesSuccessAction = articles => ({
  type: GET_ALL_ARTICLES_SUCCESS,
  payload: [articles],
});

export const getAllArticlesFailureAction = error => ({
  type: GET_ALL_ARTICLES_FAILURE,
  payload: error,
});

export const getOneArticleRequestAction = artSlug => ({
  type: GET_ONE_ARTICLE_REQUEST,
  payload: artSlug,
});

export const getOneArticleSuccessAction = article => ({
  type: GET_ONE_ARTICLE_SUCCESS,
  payload: article,
});

export const getOneArticleFailureAction = error => ({
  type: GET_ONE_ARTICLE_FAILURE,
  payload: error,
});

export const editArticleAction = article => ({
  type: EDIT_ARTICLE,
  payload: article,
});

export const updateArticleAction = article => ({
  type: UPDATE_ARTICLE,
  payload: article,
});

export const deleteArticleAction = art_slug => ({
  type: DELETE_ARTICLE,
  payload: art_slug,
});
