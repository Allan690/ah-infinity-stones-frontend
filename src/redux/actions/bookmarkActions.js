import {
  BOOKMARK_REQUEST,
  BOOKMARK_SUCCESS,
  BOOKMARK_ERROR,
} from './actionTypes';

export const bookmarkRequest = slug => ({
  type: BOOKMARK_REQUEST,
  slug,
});

export const bookmarkSuccess = data => ({
  type: BOOKMARK_SUCCESS,
  data,
});

export const bookmarkError = error => ({
  type: BOOKMARK_ERROR,
  error,
});
