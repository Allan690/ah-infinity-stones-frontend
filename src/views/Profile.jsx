import React from "react";
import Button from "../components/Profile/containers/Button";
import ProfileItem from "../components/Profile/containers/ProfileItem";
import LoadingProfile from "../components/Profile/containers/LoadingProfile";
import {
  firstname,
  lastname,
  birthday,
  bio,
  followers,
  image
} from "../services/profile.service";
import ProfileForm from "../components/Profile/containers/ProfileForm";
import "../components/Profile/Profile.scss";

let App = () => (
  <section>
    <Button />
    <LoadingProfile />
    <ProfileItem />
    <hr />
    <ProfileForm
      firstname={firstname}
      lastname={lastname}
      birthday={birthday}
      bio={bio}
      followers={followers}
      image={image}
    />
  </section>
);

export default App;
