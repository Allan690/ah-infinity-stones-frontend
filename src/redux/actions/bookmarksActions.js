import { REQUEST_BOOKMARKS, BOOKMARKS_SUCCESS, BOOKMARKS_FAILURE } from './actionTypes';

export const requestBookmarks = () => ({
  type: REQUEST_BOOKMARKS,
});

export const bookmarksSuccess = data => ({
  type: BOOKMARKS_SUCCESS,
  data,
});

export const bookmarksFailure = error => ({
  type: BOOKMARKS_FAILURE,
  error,
});
