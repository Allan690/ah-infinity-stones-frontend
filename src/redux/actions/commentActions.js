import { GET_COMMENTS, POST_COMMENT, GET_TOPIC } from './actionTypes';

export const getComments = () => ({
  type: GET_COMMENTS,
});

export const postComment = postData => ({
  type: POST_COMMENT,
  payload: postData,
});

export const getTopic = () => ({
  type: GET_TOPIC,
});
