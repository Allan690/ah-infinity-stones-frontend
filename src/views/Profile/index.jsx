import React, { Component } from 'react';
import { connect } from 'react-redux';
import Button from '../../components/Profile/containers/Button';
import ProfileItem from '../../components/Profile/containers/ProfileItem';
import LoadingProfile from '../../components/Profile/containers/LoadingProfile';
import {
  firstname,
  lastname,
  birthday,
  bio,
  followers,
  image,
} from '../../services/profile.service';
import ProfileForm from '../../components/Profile/containers/ProfileForm';
import { getProfile } from '../../redux/actions/profileActions';

class App extends Component {
  componentWillMount() {
    this.props.getProfile();
  }
  render() {
    return (
      <div>
        <div className='container'>
          <div className='row'>
            <div className='col-sm-12'>
              <Button />
              <hr />
              <h4 className='theme'>My Profile</h4>
              <LoadingProfile />
              <ProfileItem />
              <hr />
            </div>

            <div className='col-sm-6' />
            <div className='col-sm-6'>
              <ProfileForm
                firstname={firstname}
                lastname={lastname}
                birthday={birthday}
                bio={bio}
                followers={followers}
                image={image}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default connect(
  null,
  { getProfile },
)(App);
