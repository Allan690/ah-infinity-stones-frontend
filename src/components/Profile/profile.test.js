import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import Profile from './containers/ProfileItem';
import * as types from '../../redux/actions/actionTypes';
import * as actions from '../../redux/actions/profileActions';
import reducer from '../../redux/reducers/profileReducer';
import Compo from './containers/Button';

describe('<Profile />', () => {
  let cmpt;
  let button;
  const initialState = {};
  const mockStore = configureStore();
  const store = mockStore(initialState);

  beforeEach(() => {
    button = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Compo />
        </BrowserRouter>
      </Provider>,
    );
    cmpt = shallow(
      <Provider store={store}>
        {' '}
        <Profile />
        {' '}
      </Provider>,
    );
  });
  it('renders 1 Profile component', () => {
    expect(cmpt).toHaveLength(1);
  });

  it('test snapshot', () => {
    expect(cmpt).toMatchSnapshot();
  });

  it('component has a section tag', () => {
    expect(cmpt.find('section'));
  });

  it('component has a strong tag', () => {
    expect(cmpt.find('strong'));
  });

  it('Button should match snapshot', () => {
    expect(button).toMatchSnapshot();
  });
});

describe('Action creators', () => {
  let payload = {};

  beforeEach(() => {
    payload = {
      firstname: 'Promaster',
      lastname: 'Paul',
      birthday: '08/09/2000',
      image: 'http://ah-infinity-stones-frontend.herokuapp.com/static/media/logo.f64cc8b5.png',
      bio: 'A famous famer in laikipia',
    };
  });

  it('should update user profile', () => {
    const expectedAction = {
      type: types.UPDATE_PROFILE,
      payload,
    };
    expect(actions.updateProfile(payload)).toEqual(expectedAction);
  });
  it('should return user profile', () => {
    const expectedAction = {
      type: types.GET_PROFILE,
    };
    expect(actions.getProfile(payload)).toEqual(expectedAction);
  });
});
describe('Reducers', () => {
  let payload = {};

  beforeEach(() => {
    payload = {
      firstname: 'James',
      lastname: 'Mwangi',
      birthday: '05/11/2001',
      image: 'http://ah-infinity-stones-frontend.herokuapp.com/static/media/logo.f64cc8b5.png',
      bio: 'James AKA Jymo the hacker',
    };
  });

  it('should receive profile data', () => {
    expect(
      reducer(undefined, {
        type: types.PROFILE_RECEIVED,
        payload,
      }),
    ).toEqual({
      item: payload,
      loading: false,
      success: true,
      user: undefined,
    });
  });

  it('should get profile data', () => {
    expect(
      reducer(undefined, {
        type: types.GET_PROFILE,
        payload,
      }),
    ).toEqual({
      loading: true,
      success: false,
      user: {},
    });
  });

  it('should update profile with new data', () => {
    expect(
      reducer(undefined, {
        type: types.UPDATE_PROFILE,
        payload,
      }),
    ).toEqual({
      isUpdated: false,
      loading: false,
      success: false,
      user: {},
    });
  });
  it('should return a response after updating profile', () => {
    expect(
      reducer(undefined, {
        type: types.PROFILE_UPDATED,
        payload,
      }),
    ).toEqual({
      isUpdated: true,
      item: payload,
      loading: false,
      success: false,
      user: {},
    });
  });
});
