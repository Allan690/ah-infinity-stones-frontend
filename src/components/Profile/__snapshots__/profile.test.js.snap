// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> Button should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Button)>
        <Button
          getProfile={[Function]}
        >
          <div>
            <p>
              Â 
            </p>
            <button
              className="button"
              type="button"
            >
              <span
                className="theme"
              >
                <i
                  className="mdi mdi-book-open-page-variant mdi-24px"
                />
              </span>
               
              My Articles
            </button>
             
            <Link
              replace={false}
              to="/bookmarks"
            >
              <a
                href="/bookmarks"
                onClick={[Function]}
              >
                <button
                  className="button"
                  type="button"
                >
                  <span
                    className="theme"
                  >
                    <i
                      className="mdi mdi-bookmark mdi-24px"
                    />
                  </span>
                   
                  Bookmarks
                </button>
              </a>
            </Link>
             
            <button
              className="button button-theme"
              onClick={[Function]}
              type="button"
            >
              <span
                className="theme"
              >
                <i
                  className="mdi mdi-reload mdi-24px"
                />
              </span>
               
              Refresh
            </button>
          </div>
        </Button>
      </Connect(Button)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`<Profile /> test snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
   
  <[object Object] />
   
</ContextProvider>
`;
